moj-node/server/repositories/search.js

```diff
@@ -29,22 +39,32 @@ function searchRepository(httpClient) {
     return results.map(searchResultFrom);
   }
 
-  async function typeAhead({ query, limit = 5 }) {
+  async function typeAhead({ query, limit = 5, prison }) {
     const esbRequest = esb
       .requestBodySearch()
       .query(
         esb
-          .multiMatchQuery(
-            ['title^10', 'category_name^5', 'series_name^5'],
-            query,
+          .boolQuery()
+          .must(
+            esb
+              .multiMatchQuery(
+                ['title^10', 'category_name^5', 'series_name^5'],
+                query,
+              )
+              .fuzziness('AUTO')
+              .prefixLength(2)
+              .operator('and'),
           )
-          .fuzziness(2)
-          .prefixLength(2)
-          .operator('and'),
+          .should([
+            esb.termQuery('prison_name', prison),
+            esb.boolQuery().mustNot([esb.existsQuery('prison_name')]),
+          ])
+          .minimumShouldMatch(1),
       )
       .size(limit)
       .timeout('15ms')
```

Comment:
Have we tried increasing the timeout to see if yields better results?
for example we get nothing after the 4th character.. maybe increasing the timeout might yield results?

Reply:
Removed the timeout, we have such a small dataset that we're likely to resolve in a reasonable time querying it's entirety. The results remain the same, it's likely we need to experiment with prefix-lengths and fuzziness.